packages/contracts/contracts/engine/V3/GenArt721CoreV3_Engine.sol

```diff
@@ -346,63 +350,53 @@ contract GenArt721CoreV3_Engine is
     }
 
     /**
-     * @notice Initializes contract.
-     * @param _tokenName Name of token.
-     * @param _tokenSymbol Token symbol.
-     * @param _randomizerContract Randomizer contract.
+     * @notice Initializes the contract with the provided `engineConfiguration`.
+     * This function should be called atomically, immediately after deployment.
+     * Only callable once. Validation on `engineConfiguration` is performed by caller.
+     * @param engineConfiguration EngineConfiguration to configure the contract with.
      * @param _adminACLContract Address of admin access control contract, to be
      * set as contract owner.
-     * @param _startingProjectId The initial next project ID.
-     * @param _autoApproveArtistSplitProposals Whether or not to always
-     * auto-approve proposed artist split updates.
-     * @param _splitProviderAddress Address to use as royalty splitter provider for the contract.
-     * @param _nullPlatformProvider Enforce always setting zero platform provider fees and addresses.
-     * @param _allowArtistProjectActivation Allow artist to activate their own projects.
-     * @dev _startingProjectId should be set to a value much, much less than
-     * max(uint248), but an explicit input type of `uint248` is used as it is
-     * safer to cast up to `uint256` than it is to cast down for the purposes
-     * of setting `_nextProjectId`.
-     */
-    constructor(
-        string memory _tokenName,
-        string memory _tokenSymbol,
-        address _renderProviderAddress,
-        address _platformProviderAddress,
-        address _randomizerContract,
-        address _adminACLContract,
-        uint248 _startingProjectId,
-        bool _autoApproveArtistSplitProposals,
-        address _splitProviderAddress,
-        bool _nullPlatformProvider,
-        bool _allowArtistProjectActivation
-    ) ERC721_PackedHashSeed(_tokenName, _tokenSymbol) {
-        _onlyNonZeroAddress(_renderProviderAddress);
-        // @dev checks on platform provider addresses performed in _updateProviderSalesAddresses
-        if (_nullPlatformProvider) {
+     */
+    function initialize(
```

Comment:
hmm, just realized for state variables that we assign in the contract, I don't see how that would be allowed in the clone pattern. I think these should probably always be set to these values in the initialize function, and not assigned in the contract state variable section (note we do set the platform provider ones to zero sometimes already below). Happy to chat more on this if you have more thoughts!
```
uint8 private _renderProviderPrimarySalesPercentage = 10;
uint8 private _platformProviderPrimarySalesPercentage = 10;
uint256 public defaultRenderProviderSecondarySalesBPS = 250;
uint256 public defaultPlatformProviderSecondarySalesBPS = 250;
```

Reply:
Good call! I moved the assignment of these into the initialize function 🙏